core/services/subscription_internal_test.go

```diff
@@ -0,0 +1,86 @@
+package services
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/onsi/gomega"
+	"github.com/smartcontractkit/chainlink/core/services/eth/mocks"
+	"github.com/stretchr/testify/mock"
+)
+
+var _ ethereum.Subscription = EthSub{}
+
+type EthSub struct {
+	err chan error
+	t   *testing.T
+}
+
+func (es EthSub) Unsubscribe() {
+	es.t.Log("unsubscribe called")
+}
+
+func (es EthSub) Err() <-chan error {
+	es.t.Log("err chan called")
+	return es.err
+}
+
+func TestSubscriptionListenToLogs(t *testing.T) {
```

Comment:
I am more confident now I see a test. Did this fail on the previous code ?

Reply:
Yes this failed on previous code - the fact that the we were updating a copy of sub.logs in the reconnect meant that the unsubscribe wouldn't work. However that is just sort of an unrelated landmine - the real fix is just to not empty the job subscription map in Disconnect here https://github.com/smartcontractkit/chainlink/pull/4342/files#diff-3aa9e1f0eaf11525321dfd8287eb6e975bfcc57899fdf2b79e2848ef48cd9b50L188